src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  const dateComponents = fromFormat.slice(0, -1).reduce((acc, key, i) => {
+    acc[key] = oldDateParts[i];
+
+    return acc;
+  }, {});
+
+  if (dateComponents['YY']) {
+    dateComponents['YYYY'] =
+      dateComponents['YY'] < 30
+        ? '20' + dateComponents['YY']
+        : '19' + dateComponents['YY'];
+  } else {
+    dateComponents['YY'] = dateComponents['YYYY'].substring(2);
```

Comment:
Similar to the previous comment, this line assumes 'YYYY' will be present in the `dateComponents`. It would be safer to check if 'YYYY' is a part of the `fromFormat` before performing this operation.